tests/shared/hooks.js

```diff
@@ -4,29 +4,28 @@ const {wrappedError} = require('../../src/utils');
 const {log} = require('../../src/logger');
 
 
-exports.remoteSwarmHook = function ({createDB = true, log = false, logDetailed = false} = {}) {
+exports.remoteSwarmHook = function ({createDB, uuid, ethereum_rpc, esrContractAddress, private_pem, public_pem, log, logDetailed} = {}) {
+    const clientArguments = {uuid, ethereum_rpc, esrContractAddress, private_pem, public_pem, log, logDetailed};
+
     before('initialize client and setup db', async function () {
         this.timeout(harnessConfigs.defaultBeforeHookTimeout);
-        this.api = await remoteSetup({createDB, log, logDetailed});
+
+        this.api = await remoteSetup({createDB, ...clientArguments});
```

Comment:
Now seeing the full impl. I think I see an issue with this approach.

Everything in `clientArguments` has a value, which could be `undefined` (the debate for that being a value or not can be held another day). That means if we don't specify a value, it won't let the `bluzelle` constructor use its default and will pass through `undefined` instead.

Thoughts?

Reply:
I'm not sure that passing `undefined` will prevent the constructor use its default. I tried this simple example and it seems like it uses default even when passed `undefined`

```
(() => {

    const args = {one: undefined, two: undefined, three: undefined};

    func()                    // 1 2 3
    func(args)             // 1 2 3

})();

function func({one = 1, two = 2, three = 3} = {}) {
    console.log(one, two, three);
}
```

Reply:
Huh. I swear I had a counter-example but upon seeing how the client handles defaults (via body, not signature), it should be safe even with `undefined`. Continuing review.

Reply:
Yeah, I double checked client code too 

```arg = arg || default```

That works with `undefined` too 
👏